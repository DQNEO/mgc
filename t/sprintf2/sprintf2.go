package main

import (
	"fmt"
	"os"
)

func receiveSliceInVariadic(a ...interface{}) {
	var i0 interface{} = a[0]
	var b []byte
	var ok bool

	b, ok = i0.([]byte)

	fmt.Printf("ok=%v\n", ok)
	fmt.Printf("b=%s,len=%d,cap=%d\n", b, len(b), cap(b))
}

func f0() {
	receiveSliceInVariadic([]byte("abc"))
}

func Write(s string) {
	os.Stdout.Write([]byte(s))
}
func f1() {
	var b string
	b = fmt.Sprintf("hello\n")
	Write(b)

	b = fmt.Sprintf("%s\n", "world")
	Write(b)

	b = fmt.Sprintf("%s\n", "world")
	Write(b)

	b = fmt.Sprintf("left %s right\n", "center")
	Write(b)

	b = fmt.Sprintf("%s center right\n", "left")
	Write(b)

	b = fmt.Sprintf("left center %s", "right\n")
	Write(b)

	b = fmt.Sprintf("%s center %s\n", "left", "right")
	Write(b)

	var i int

	i = 123
	b = fmt.Sprintf("123=%d\n", i)
	Write(b)

	i = 4567
	b = fmt.Sprintf("%s=%d\n", "4567", i)
	Write(b)
}

func f2() {
	var b string
	b = fmt.Sprintf("pushq %%rax\n")
	Write(b)
}

func main() {
	fmt.Printf("--- f0 ---\n")
	f0()
	fmt.Printf("--- f1 ---\n")
	f1()
	fmt.Printf("--- f2 ---\n")
	f2()
}
