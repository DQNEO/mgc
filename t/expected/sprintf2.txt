--- f0 ---
ok=true
b=abc,len=3,cap=3
--- f1 ---
hello
world
world
left center right
left center right
left center right
left center right
123=123
4567=4567
--- f2 ---
pushq %rax
